src/oscar/apps/order/utils.py

```diff
@@ -258,3 +260,22 @@ def record_voucher_usage(self, order, voucher, user):
         Updates the models that care about this voucher.
         """
         voucher.record_usage(order, user)
+
+
+def create_invoice(order):
+    """
+    To create `Invoice` instance, we should have at least one
+    instance of `LegalEntity` with `LegalEntityAddress`.
+
+    Some platforms may have couple `LegalEntity`s (with couple
+    `LegalEntityAddress`s). In this case needed instances should be
+    selected based on order (ordered products).
+
+    """
+    legal_entity = LegalEntity.objects.first()
+    if legal_entity and legal_entity.has_addresses:
+
+        invoice_number = int(Invoice.get_last_invoice_number()) + 1
```

Comment:
This is not safe from race conditions. if the `number` is based on `id` then it should probably be a property rather than a separate field?